### ISSUE TITLE
NoSuchMethodError: The getter 'flow' was called on null.

### ISSUE THREAD

**Issue by purgeyao** [Milad-Akarie/auto_route_library#499]
auto_route_generator: ^0.6.9
auto_route: ^0.6.9

perform
```
flutter packages pub run build_runner build
```
error

```
flutter packages pub run build_runner build
[INFO] Generating build script...
[INFO] Generating build script completed, took 492ms

[INFO] Creating build script snapshot......
[INFO] Creating build script snapshot... completed, took 12.7s

[INFO] There was output on stdout while compiling the build script snapshot, run with `--verbose` to see it (you will need to run a `clean` first to re-snapshot).

[INFO] Initializing inputs
[INFO] Building new asset graph...
[INFO] Building new asset graph completed, took 629ms

[INFO] Checking for unexpected pre-existing outputs....
[INFO] Checking for unexpected pre-existing outputs. completed, took 1ms

[INFO] Running build...
[INFO] 1.0s elapsed, 110/113 actions completed.
[INFO] 2.1s elapsed, 146/147 actions completed.
[INFO] 3.1s elapsed, 146/147 actions completed.
[INFO] 4.1s elapsed, 146/147 actions completed.
[SEVERE] auto_route_generator:autoRouteGenerator on lib/generated/i18n.dart:

NoSuchMethodError: The getter 'flow' was called on null.
Receiver: null
Tried calling: flow
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:54:5)
#1      ResolverVisitor.startNullAwarePropertyAccess (package:analyzer/src/generated/resolver.dart:758:23)
#2      ResolverVisitor.visitPropertyAccess (package:analyzer/src/generated/resolver.dart:1730:5)
#3      PropertyAccessImpl.accept (package:analyzer/src/dart/ast/ast.dart:8398:49)
#4      BinaryExpressionResolver._resolveEqual (package:analyzer/src/dart/resolver/binary_expression_resolver.dart:111:10)
#5      BinaryExpressionResolver.resolve (package:analyzer/src/dart/resolver/binary_expression_resolver.dart:58:7)
#6      ResolverVisitor.visitBinaryExpression (package:analyzer/src/generated/resolver.dart:897:31)
#7      BinaryExpressionImpl.accept (package:analyzer/src/dart/ast/ast.dart:954:49)
#8      BinaryExpressionResolver._resolveLogicalOr (package:analyzer/src/dart/resolver/binary_expression_resolver.dart:224:11)
#9      BinaryExpressionResolver.resolve (package:analyzer/src/dart/resolver/binary_expression_resolver.dart:63:7)
#10     ResolverVisitor.visitBinaryExpression (package:analyzer/src/generated/resolver.dart:897:31)
#11     BinaryExpressionImpl.accept (package:analyzer/src/dart/ast/ast.dart:954:49)
#12     ResolverVisitor.visitAssertInitializer (package:analyzer/src/generated/resolver.dart:856:21)
#13     AssertInitializerImpl.accept (package:analyzer/src/dart/ast/ast.dart:517:49)
#14     AstResolver.resolve (package:analyzer/src/summary2/ast_resolver.dart:85:10)
#15     ConstructorInitializerResolver._initializers (package:analyzer/src/summary2/constructor_initializer_resolver.dart:68:20)
#16     ConstructorInitializerResolver._constructor (package:analyzer/src/summary2/constructor_initializer_resolver.dart:54:5)
#17     ConstructorInitializerResolver.resolve (package:analyzer/src/summary2/constructor_initializer_resolver.dart:31:11)
#18     LibraryBuilder.resolveConstructors (package:analyzer/src/summary2/library_builder.dart:240:53)
#19     Linker._resolveConstructors (package:analyzer/src/summary2/link.dart:210:15)
#20     Linker._buildOutlines (package:analyzer/src/summary2/link.dart:95:5)
#21     Linker.link (package:analyzer/src/summary2/link.dart:76:5)
#22     link (package:analyzer/src/summary2/link.dart:32:10)
#23     LibraryContext.load2.loadBundle (package:analyzer/src/dart/analysis/library_context.dart:215:24)
#24     LibraryContext.load2.loadBundle.<anonymous closure> (package:analyzer/src/dart/analysis/library_context.dart:123:26)
#25     SetMixin.forEach (dart:collection/set.dart:142:30)
#26     LibraryContext.load2.loadBundle (package:analyzer/src/dart/analysis/library_context.dart:122:32)
#27     LibraryContext.load2.loadBundle.<anonymous closure> (package:analyzer/src/dart/analysis/library_context.dart:123:26)
#28     SetMixin.forEach (dart:collection/set.dart:142:30)
#29     LibraryContext.load2.loadBundle (package:analyzer/src/dart/analysis/library_context.dart:122:32)
#30     LibraryContext.load2.loadBundle.<anonymous closure> (package:analyzer/src/dart/analysis/library_context.dart:123:26)
#31     SetMixin.forEach (dart:collection/set.dart:142:30)
#32     LibraryContext.load2.loadBundle (package:analyzer/src/dart/analysis/library_context.dart:122:32)
#33     LibraryContext.load2.loadBundle.<anonymous closure> (package:analyzer/src/dart/analysis/library_context.dart:123:26)
#34     SetMixin.forEach (dart:collection/set.dart:142:30)
#35     LibraryContext.load2.loadBundle (package:analyzer/src/dart/analysis/library_context.dart:122:32)
#36     LibraryContext.load2.<anonymous closure> (package:analyzer/src/dart/analysis/library_context.dart:262:17)
#37     PerformanceLog.run (package:analyzer/src/dart/analysis/performance_logger.dart:32:15)
#38     LibraryContext.load2 (package:analyzer/src/dart/analysis/library_context.dart:260:12)
#39     new LibraryContext (package:analyzer/src/dart/analysis/library_context.dart:77:5)
#40     AnalysisDriver._createLibraryContext.<anonymous closure> (package:analyzer/src/dart/analysis/driver.dart:1538:27)
#41     _rootRun (dart:async/zone.dart:1354:13)
#42     _CustomZone.run (dart:async/zone.dart:1258:19)
#43     _runZoned (dart:async/zone.dart:1788:10)
#44     runZoned (dart:async/zone.dart:1710:10)
#45     NullSafetyUnderstandingFlag.enableNullSafetyTypes (package:analyzer/dart/element/null_safety_understanding_flag.dart:42:12)
#46     AnalysisDriver._createLibraryContext (package:analyzer/src/dart/analysis/driver.dart:1536:33)
#47     AnalysisDriver._computeUnitElement.<anonymous closure> (package:analyzer/src/dart/analysis/driver.dart:1464:28)
#48     PerformanceLog.run (package:analyzer/src/dart/analysis/performance_logger.dart:32:15)
#49     AnalysisDriver._computeUnitElement (package:analyzer/src/dart/analysis/driver.dart:1462:20)
#50     AnalysisDriver.performWork (package:analyzer/src/dart/analysis/driver.dart:1020:34)
#51     AnalysisDriverScheduler._run (package:analyzer/src/dart/analysis/driver.dart:2013:24)
<asynchronous suspension>


[WARNING] No actions completed for 16.9s, waiting on:
  - auto_route_generator:autoRouteGenerator on lib/router/route_map.dart


```

**Comment by Milad-Akarie** [comment]
@purgeyao  please consider migrating to auto_route 1.0+ as the old versions are no longer supported

**Comment by theweiweiway** [comment]
Closing due legacy generator not being supported anymore - please open a new issue for AutoRoute 2.x if you are still having problems!
